### ISSUE TITLE
columnspacing not working for amstock chart and also for series chart 

### ISSUE THREAD

**Issue by savita1378** [amcharts/amcharts3#128]

**Comment by martynasma** [comment]
Could you please provide more details, or a demo?

**Comment by savita1378** [comment]
Hi,
Can you also tell me is amchart compatible with angular 2?

**Comment by martynasma** [comment]
Yes, it is. We have an Angular2 plugin and usage instructions here:

https://github.com/amcharts/amcharts3-angular2

**Comment by savita1378** [comment]
Hi Team ,
Can you please suggest  me that in amchart that can we upload partial data
when scrolling back in amchart as when required.

Like we have data for 6 month and it is showing in amchart and if we
required to see data for previous 12 months and we scoll back then is it
possible in amchart that it will fetch data on scrolling back

Thanks,
Savita Dabral

**Comment by martynasma** [comment]
Well, there is no built-in functionality for that.

However, you can watch the `zoomed` event on the chart and dynamically update `dataProvider`.

Here's a demo for that for Stock Chart:

https://www.amcharts.com/kbase/incremental-on-demand-data-loading-for-stock-chart/

Hopefully, it gets you on the right track.

**Comment by savita1378** [comment]
Hi,
Can you also  tell me that can we modify the previous value/data in amchart
after loading of the chart

**Comment by martynasma** [comment]
Sure.

To modify the data on the Stock Chart, you need to modify `dataProvider` of the respective data set, as well as call `validateData()`. E.g.:

```
chart.dataSets[0].dataProvider = newData;
chart.valdiateData();
```

Hope this helps.

**Comment by savita1378** [comment]
Hi Team,
Can you tell me any mouse drag event in amchart .

On Fri, Sep 29, 2017 at 11:57 AM, Martynas Majeris <notifications@github.com

**Comment by martynasma** [comment]
Could you please be more specific? Are you referring to the cursor being dragged to zoom?

**Comment by savita1378** [comment]
Hi Team,
I am facing flickering (on mouse hover)  in chart when we insert real time
data. can you please tell how to stop flickering on mouse hover
Thanks,
Savita Dabral

On Fri, Oct 13, 2017 at 10:10 AM, Martynas Majeris <notifications@github.com

**Comment by martynasma** [comment]
Again, if you could provide a sample of such case, that would certainly help.

**Comment by savita1378** [comment]
can you check this example  http://jsfiddle.net/amcharts/s6BnP/


On Thu, Oct 26, 2017 at 10:41 AM, Martynas Majeris <notifications@github.com

**Comment by martynasma** [comment]
Thanks.

While there's not a lot that can be done about the flickering, I found that setting cursor's position to "mouse", helps a little:

```
chart.chartCursorSettings.cursorPosition = "mouse";
```

I hope you find this useful.

**Comment by savita1378** [comment]
Hi Team,
Thanks for the suggestion.
Can you please also tell me that how to add white empty space in the chart.
Because all the charts in amcharts forms at the right end . we need some
empty space right side.


here is the example:
https://www.amcharts.com/kbase/controlling-category-axis-start-end-points-adding-empty-data-points/


Can we have some other way to do this as this is not the ideal way to add
space

Thanks,
Savita dabral

**Comment by martynasma** [comment]
You can use an automatic plugin to do that for you:

https://github.com/amcharts/tools/tree/master/datePadding
